app/models/miq_task.rb

```diff
@@ -59,8 +61,8 @@ def check_associations
   def update_status(state, status, message)
     status = STATUS_ERROR if status == STATUS_EXPIRED
     _log.info("Task: [#{id}] [#{state}] [#{status}] [#{message}]")
-    attributes = {:state => state, :status => status, :message => message}
-    attributes[:started_on] = Time.now.utc if state == STATE_ACTIVE && started_on.nil?
+    attributes = {:status => status, :message => message}
+    attributes = attributes_hash_when_state_change(state, attributes)
```

Comment:
@Fryguy agreed, it does look artificial and  reasons for that: attempt to avoid repeating the same code